transport/src/main/java/io/netty/channel/pool/AbstractChannelPoolMap.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.channel.pool;
+
+import io.netty.util.internal.PlatformDependent;
+import io.netty.util.internal.ReadOnlyIterator;
+
+import java.util.Iterator;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentMap;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+/**
+ * A skeletal {@link ChannelPoolMap} implementation. To find the right {@link ChannelPool}
+ * the {@link Object#hashCode()} and {@link Object#hashCode()} is used.
```

Comment:
You added ``{@link Object#hashCode()}``, I guess you meant ``{@link Object#equals()}``